src/stores/ReadTrainerStore.js

```diff
@@ -29,7 +29,8 @@ class ReadTrainerStore extends SettingsSaver {
       maxDictionarySize: dictionary.wordFrequency.size,
       types: dictionary.typesToIncludeByDefault,
       useUserDictionary: false,
-      userDictionary: observable.array(),
+      userDictionaryWords: observable.array(),
```

Comment:
How about making this an observable.map() and storing the frequencies inside? It would simplify everything else below.

Reply:
Sure. I can re-work the implementation. It will take me a bit longer than usual. With some bad timing, my iMac hard drive at home has gone out and I am waiting on it to get fixed. I can work on the change next week (or perhaps stay a little later at work and work on it).

Reply:
I finally got my iMac fixed and restored. What do you think about the implementation now?

Reply:
Glad to hear about the Mac :)

The code looks great -- thanks! Merged it now and will try to do a release soon.